### PR TITLE
properly mock rds name update

### PR DESCRIPTION
**Pull request by ehowe**

### CODE REVIEW

File: lib/fog/aws/requests/rds/modify_db_instance.rb [hunk 1]
```diff
@@ -61,6 +61,9 @@ def modify_db_instance(db_name, apply_immediately, options={})
               #else
               #  modified_server = server["PendingModifiedValues"].merge!(options) # it appends
               #end
+              if options["NewDBInstanceIdentifier"]
+                options["DBInstanceIdentifier"] = options.delete("NewDBInstanceIdentifier")
```
Comment (lanej) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: this should go in `PendingModifiedValues`, so it won't take effect until a reload.

see https://github.com/fog/fog-aws/blob/master/lib/fog/aws/requests/rds/describe_db_instances.rb#L58

Comment (ehowe) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: That is not the API behavior.  This attribute takes effect half instantly.  The ID you query is the old id, but the ID returned is the new one.  This never goes into `PendingModifiedValues`

Comment (lanej) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: image

Comment (lanej) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: shows in `PendingModifiedValues`

```xml
<DescribeDBInstancesResponse xmlns="http://rds.amazonaws.com/doc/2014-10-31/">
  <DescribeDBInstancesResult>
    <DBInstances>
      <DBInstance>
        <MonitoringInterval>0</MonitoringInterval>
        <BackupRetentionPeriod>0</BackupRetentionPeriod>
        <DBInstanceStatus>available</DBInstanceStatus>
        <MultiAZ>false</MultiAZ>
        <VpcSecurityGroups/>
        <DBInstanceIdentifier>fog-test-0649</DBInstanceIdentifier>
        <PreferredBackupWindow>04:25-04:55</PreferredBackupWindow>
        <PreferredMaintenanceWindow>sat:07:49-sat:08:19</PreferredMaintenanceWindow>
        <AvailabilityZone>us-east-1e</AvailabilityZone>
        <DomainMemberships/>
        <CACertificateIdentifier>rds-ca-2015</CACertificateIdentifier>
        <ReadReplicaDBInstanceIdentifiers/>
        <Engine>mysql</Engine>
        <PendingModifiedValues>
          <DBInstanceIdentifier>fog-lanej</DBInstanceIdentifier>
        </PendingModifiedValues>
        <LicenseModel>general-public-license</LicenseModel>
        <KmsKeyId>arn:aws:kms:us-east-1:915445820265:key/9363058e-fd3f-4971-b6a9-6e8170b35172</KmsKeyId>
        <DbiResourceId>db-XAGELCI75QEPEI6OGKAWDODN4U</DbiResourceId>
        <DBParameterGroups>
          <DBParameterGroup>
            <ParameterApplyStatus>in-sync</ParameterApplyStatus>
            <DBParameterGroupName>default.mysql5.6</DBParameterGroupName>
          </DBParameterGroup>
        </DBParameterGroups>
        <Endpoint>
          <Port>3306</Port>
          <Address>fog-test-0649.cbffcxs6yujt.us-east-1.rds.amazonaws.com</Address>
        </Endpoint>
        <EngineVersion>5.6.23</EngineVersion>
        <OptionGroupMemberships>
          <OptionGroupMembership>
            <OptionGroupName>default:mysql-5-6</OptionGroupName>
            <Status>in-sync</Status>
          </OptionGroupMembership>
        </OptionGroupMemberships>
        <PubliclyAccessible>true</PubliclyAccessible>
        <DBSecurityGroups>
          <DBSecurityGroup>
            <Status>active</Status>
            <DBSecurityGroupName>default</DBSecurityGroupName>
          </DBSecurityGroup>
        </DBSecurityGroups>
        <AutoMinorVersionUpgrade>true</AutoMinorVersionUpgrade>
        <StorageType>standard</StorageType>
        <DbInstancePort>0</DbInstancePort>
        <InstanceCreateTime>2015-08-05T17:47:20.889Z</InstanceCreateTime>
        <AllocatedStorage>5</AllocatedStorage>
        <StorageEncrypted>true</StorageEncrypted>
        <CopyTagsToSnapshot>false</CopyTagsToSnapshot>
        <DBInstanceClass>db.m3.medium</DBInstanceClass>
        <MasterUsername>foguser</MasterUsername>
      </DBInstance>
    </DBInstances>
  </DescribeDBInstancesResult>
  <ResponseMetadata>
    <RequestId>5d9ed1a8-42a3-11e5-8537-afbd649eaded</RequestId>
  </ResponseMetadata>
</DescribeDBInstancesResponse>
```

Comment (ehowe) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: The ui does not accurately reflect the api behavior

Comment (ehowe) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: Did you update that via the ui or API?

Comment (lanej) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: i believe the parser is broken.

Comment (ehowe) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: definitely broken parser.  seems that anything in `PendingModifiedValues` gets assigned to the model by the parser.

Comment (lanej) [inline lib/fog/aws/requests/rds/modify_db_instance.rb]: i think it's just keys assigned farther up in the case statement.

**Comment by geemus** [comment]
Thanks!

**Pull request merged**